### PR TITLE
sql: fixed diff tool add missing columns not working

### PR DESCRIPTION
**Pull request by mnovelodou**
Previously, add missing column for diff tool stop working for
existing and implemented tables
This was inadequate because this code should be able to add new
columns for all tables in pg_catalog
To address this, this patch fixes the bug for adding columns and
now can implement new columns at tables that it couldn't add before

Release note: None

### CODE REVIEW

File: pkg/sql/pg_metadata_test.go [hunk 1]
```diff
@@ -1080,16 +1165,39 @@ func (v *pgCatalogCodeVisitor) findSchemaIndex(call *ast.CallExpr) int {
 	return index
 }
 
+func (v *pgCatalogCodeVisitor) findPopulateFunc(call *ast.CallExpr) int {
```
Comment (RichardJCai) [inline pkg/sql/pg_metadata_test.go:1168]: Can you comment this function too?

File: pkg/sql/pg_metadata_test.go [hunk 2]
```diff
@@ -826,21 +836,23 @@ func getSortedDefKeys(tableDefs map[string]string) []string {
...
 	pgCode := &pgCatalogCode{
-		fixableTables:   make(map[string]struct{}),
-		addRowPositions: make(map[string]addRowPositionList),
-		schemaParam:     -1, // This value will be calculated later and once
-		srcFile:         srcFile,
-		t:               t,
+		fixableTables:       make(map[string]struct{}),
+		addRowPositions:     make(mappedRowPositions),
+		functionsWithAddRow: make(mappedRowPositions),
+		schemaParam:         -1, // This value will be calculated later and once
```
[the remark this answers is not in the document]
Comment (mnovelodou) [reply pkg/sql/pg_metadata_test.go:852]: Done.

File: pkg/sql/pg_metadata_test.go [hunk 3]
```diff
@@ -1230,11 +1338,48 @@ func TestPGCatalog(t *testing.T) {
...
+
 	// rewrite undefinedTables.
 	validateUndefinedTablesField(t)
 	validateTableDefsField(t)
 	validateVirtualSchemaTable(t)
 	validateFunctionNames(t)
+	validatePGCatalogCodeParser(t)
+}
+
+// validatePGCatalogCodeParser will test that pg_catalog.go can be fixed
```
[the remark this answers is not in the document]
Comment (mnovelodou) [reply pkg/sql/pg_metadata_test.go:1350]: Done.

**Review by RichardJCai — APPROVED** [review]
LGTM after comment fix

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/64606#-:-M_Mpi7U8ryzTCSUJOvd:bmc2ei1)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @mnovelodou and @RichardJCai)

___
*[pkg/sql/pg_metadata_test.go, line 1174 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/64606#-M_MpZwg5ADbWYvZ0aac:-M_MpZwg5ADbWYvZ0aad:b-bnjugx) ([raw file](https://github.com/cockroachdb/cockroach/blob/b508586368b77611a5ce4bf874cd70f9d28d4b98/pkg/sql/pg_metadata_test.go#L1174)):*
> ```Go
> // function, which later will be used to find "addRow" calls.
> // The difference is: If populate func is defined there, then that function
> // definition have the addRow calls (And we can match these addRow calls with
> ```

have -> has 

(definition has the addRow calls)

**Pull request merged**